airflow/dag_processing/processor.py

```diff
@@ -513,7 +513,12 @@ def update_import_errors(session: Session, dagbag: DagBag) -> None:
         """
         # Clear the errors of the processed files
         for dagbag_file in dagbag.file_last_changed:
-            session.query(errors.ImportError).filter(errors.ImportError.filename == dagbag_file).delete()
+            if dagbag_file.endswith(".zip"):
```

Comment:
IIRC the DAG parsing code (`correct_if_zipped`?) uses `zipfile.is_zipfile` and accepts zip files without the `.zip` suffix, so this probably should do the same.

Actually, I wonder if it’s worthwhile to separate the file’s display (which includes the part inside a zip) and the actual path to avoid this check altogether.

Reply:
Ah, thanks! I think the easiest fix is to just use `startswith` for all of them.